pkg/provisioner/ironic/clients/client.go

```diff
@@ -25,20 +27,33 @@ type TLSConfig struct {
 }
 
 func updateHTTPClient(client *gophercloud.ServiceClient, tlsConf TLSConfig) (*gophercloud.ServiceClient, error) {
-	tlsInfo := transport.TLSInfo{
-		TrustedCAFile:      tlsConf.TrustedCAFile,
-		InsecureSkipVerify: tlsConf.InsecureSkipVerify,
-	}
 	if _, err := os.Stat(tlsConf.TrustedCAFile); err != nil {
 		if os.IsNotExist(err) {
-			tlsInfo.TrustedCAFile = ""
+			tlsConf.TrustedCAFile = ""
 		} else {
 			return client, err
 		}
 	}
-	tlsTransport, err := transport.NewTransport(tlsInfo, tlsConnectionTimeout)
-	if err != nil {
-		return client, err
+	/* #nosec G402*/
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: tlsConf.InsecureSkipVerify,
+	}
+	if tlsConf.TrustedCAFile != "" {
+		certs := x509.NewCertPool()
+		pemData, err := ioutil.ReadFile(tlsConf.TrustedCAFile)
+		if err != nil {
+			return client, fmt.Errorf("Failed to open CA certificate %s: %v", tlsConf.TrustedCAFile, err)
+		}
+		ok := certs.AppendCertsFromPEM(pemData)
+		if !ok {
+			return client, fmt.Errorf("Failed to  append CA certificate %s", tlsConf.TrustedCAFile)
+		}
+		tlsConfig.RootCAs = certs
+	}
+
+	tlsTransport := &http.Transport{
+		TLSClientConfig:   tlsConfig,
+		DisableKeepAlives: true,
```

Comment:
Couldn't we just set this one field before passing the Transport to http.Client, without necessarily reimplementing the convenience functions that we were importing from etcd?

Reply:
@zaneb I am not sure if etcd transport has the DisableKeepAlive field? That's why we are re implementing it. I might be wrong. Please provide some pointers if I missed something.

Reply:
It doesn't as far as I can see, but `NewTransport()` returns a `http.Transport` object (`tlsTransport`) that we then store in the `http.Client` struct. We just have to set that field while we have custody of the struct.

Reply:
@zaneb actually you are right. Does it look good now?